### DISCUSSION TITLE
Can we make `useUrlSearchParams` be `vue-router` aware?

### DISCUSSION THREAD

**Discussion by isgj** [vueuse/vueuse#550] [Q&A] [answered]
I have a page that uses `useUrlSearchParams`, everything works fine, the url is updated correctly. But when I click to a link and go back, the initial url is used, `vue router` doesn't know about the urls that were updated by `useUrlSearchParams`. Can I somehow make them work together?

**Comment by antfu** [comment, marked as answer]
Tried [@vueuse/router](https://vueuse.org/add-ons.html#router-vueuse-router)?

**Reply by isgj** [reply]
Thanks, the only difference is `useURLSearchParams` has all the query strings while `useRouteQuery` binds to one value. But of course the result can be achieved also with that.

PS: I should've guess this from the naming, it makes sense :smile:
